### PR TITLE
feat: add glint plugin fixes to reporter

### PR DESCRIPTION
**Pull request by cafreeman**
Hooks glint plugins into the Rehearsal reporting system.

### CODE REVIEW

File: packages/plugins/src/plugins/glint-report.plugin.ts [hunk 1]
```diff
@@ -37,10 +41,32 @@ export class GlintReportPlugin implements Plugin<PluginOptions> {
     return Promise.resolve([fileName]);
   }
 
+  getDiagnostics(service: Service, fileName: string): DiagnosticWithContext[] {
+    const languageService = service.getLanguageService();
+    const program = languageService.getProgram()!;
+    const checker = program.getTypeChecker();
+
+    const diagnostics = service.getDiagnostics(fileName);
```
Comment (chadhietala) [inline packages/plugins/src/plugins/glint-report.plugin.ts:49]: I think you want to go through this https://github.com/rehearsal-js/rehearsal-js/blob/master/packages/codefixes/src/get-diagnostics.ts#L26

Comment (cafreeman) [reply packages/plugins/src/plugins/glint-report.plugin.ts]: Good call, fixed.

**Review by egorio — DISMISSED** [review]

**Comment by lynchbomb** [comment]
@cafreeman we should validate the glint comments don't cause issue with the various @rehearsal/reporter formatters eg. SARIF, sonar

For this PR could you expand the tests in @rehearsal/reporter to include examples with glint comments in the report

**Comment by lynchbomb** [comment]
otherwise LGTM

**Comment by cafreeman** [comment]
> @cafreeman we should validate the glint comments don't cause issue with the various @rehearsal/reporter formatters eg. SARIF, sonar
> 
> For this PR could you expand the tests in @rehearsal/reporter to include examples with glint comments in the report

Discussed in Slack. Since the glint comments actually show up as regular TS errors as far as the report is concerned, nothing really changes in the formatters here so there isn't anything to test

**Review by lynchbomb — APPROVED** [review]

**Pull request merged**